File: src/com/jetbrains/idear/actions/recognition/RunActionRecognizer.java
package com.jetbrains.idear.actions.recognition;

import org.jetbrains.annotations.NotNull;

//runs only selected configuration
public class RunActionRecognizer implements ActionRecognizer {

    @Override
    public boolean isMatching(@NotNull String sentence) {
        return sentence.contains("run");
    }

    @Override
    public ActionCallInfo getActionInfo(@NotNull String sentence) {
        return new ActionCallInfo("Run", null);
    }
}

File: src/main/java/org/openasr/idear/actions/recognition/RunActionRecognizer.kt
package org.openasr.idear.actions.recognition

import com.intellij.openapi.actionSystem.DataContext
import com.intellij.openapi.editor.impl.EditorComponentImpl
import java.awt.Component

//runs only selected configuration
class RunActionRecognizer : ActionRecognizer {
    override fun isSupported(dataContext: DataContext, component: Component?) = component is EditorComponentImpl
    override fun isMatching(utterance: String) = "run" in utterance
    override fun getActionInfo(utterance: String, dataContext: DataContext) = ActionCallInfo("Run")
}
